pwndbg/dbg/gdb.py

```diff
@@ -738,7 +738,11 @@ def module_section_locations(self) -> List[Tuple[int, int, str, str]]:
     def main_module_name(self) -> str | None:
         # Can GDB ever return a different value here from what we'd get with
         # `info files`, give or take a "remote:"?
-        return gdb.current_progspace().filename
+        if not self.alive():
+            return gdb.current_progspace().filename
+
+        exe = gdb.execute("info proc exe", to_string=True)
```

Comment:
Fwiw, just for information, it seems GDB fetches this via `/proc/$pid/exe`:

https://github.com/bminor/binutils-gdb/blob/gdb-15.1-release/gdb/infcmd.c#L3024-L3028

https://github.com/bminor/binutils-gdb/blob/gdb-15.1-release/gdb/infcmd.c#L2960-L2972


https://github.com/bminor/binutils-gdb/blob/gdb-15.1-release/gdb/linux-tdep.c#L810-L821

https://github.com/bminor/binutils-gdb/blob/gdb-15.1-release/gdb/linux-tdep.c#L882-L891

Reply:
make sense, but creating a custom implementation that simulates this behavior (for remote and local) would be worse imo (way to complex for what it's worth)
therefore i would believe that `info proc exe`, is still the preferred solution